README.md
# devC_esp-idf
DevC++ - IDE with ESP32 esp-idf support  

![alt tag](https://raw.githubusercontent.com/ESP32DE/devC_esp-idf/master/devC%20with%20ESP-IDF%20Support.jpg)

# Credit : 
  http://www.bloodshed.net/devcpp.html
    
# visit
  http://esp32.com/viewtopic.php?f=17&t=294&p=1270#p1270

  http://esp32.de

# donate
  ...no this is fun :)  







SOURCE 5.11/NEWS.txt

Version 5.11 portable version 
  v0.95 esp-idf rudi ;-) - 03 Oktober 2016

  - add ESP32 esp-idf support
  - make targets
  - debug / - uart version ( not pre released ) 
  - debug / - tcp version  ( not pre release  ) 
  - remove donat link
  - add github link
  - remove donat image
  - add author for esp-idf version 
  - pre release 5.11 v0.95 esp-idf version on 03 Oktober 2016
  
  - knowed: 
    a) do not make target without open a ESP32 esp-idf project
       there is just in time no check for valid call
       be sure you have open/created a valid ESP esp-idf project

    b) you need to setup the Makefile for each Project by yourself
       standard is, that in each new project the Makefile is done by DevC++
       You must set up in project option "do not create Makefile" and to use own Makefile
       
    c) this is prerelease, be shure you do not use it in productive doings :)
    
    

Version 5.11 - 27 April 2015

- Fixed crash related to double clicking on a compiler error when a selection was made.
- Upgraded the default compiler to TDM-GCC 4.9.2.
- Improved startup speed.
- Fixed Abort Compilation button not working anymore.
- Fixed crash in TCppParser.CheckForTypedefStruct.
- Fixed crash in TCppParser.HandleEnum.
- Fixed some typos in the English translation (thanks to Hiro5).
- Updated the Catalan translation (thanks to Hiro5).
- Updated the Czech translation (thanks to tringi).
- Fixed some hickups in the build process of Dev-C++ itself.

Version 5.10 - 11 March 2015

- Improved startup speed.
- Removed splash window.
- Temporarily removed the web update window.
- Rewrote the LangCheck tool that can be used to validate translation files.
- Improved indent guides painting and positioning.
- Removed full screen information bar.
- Added GUI option for maximum line length option to AStyle > Formatter Options.
- Fixed parameter mismatch in Chinese translation.
- Fixed memory leak in Edit > Unindent.
- Improved behaviour of Edit > Toggle Comment.
- Improved opening speed of Tools > Compiler Options.

Version 5.9.2 - 1 Februari 2015

- Changed Format Current File shortcut from Shift+F to Shift+Ctrl+A.
- Removed/Changed all default Alt+(Key) shortcuts because they interfere with Alt menu navigation.
- Assigned some new default shortcuts like Ctrl+B (Open Containing Folder) and F2 (Rename File).
- Fixed Show Makefile being executable when no compiler set is configured.

Version 5.9.1 - 31 Januari 2015

- Updated Tools >> Edit Shortcuts.
- Removed Ctrl+Alt combination in default shortcuts to ensure that the AltGr button keeps functioning:
http://blogs.msdn.com/b/oldnewthing/archive/2004/03/29/101121.aspx

- Fixed formatter options command line not showing up after first launch.
- Added source LaTeX of testing document DevTest.pdf.
- Widened the Menu >> Search >> Find window in order to accomodate longer translations.
- Fixed a crash when launching for the first time with no compilers available.
- Added last selection arrow to Find Results.
- Fixed out of bounds error in TCppParser.HandleMethod.

Version 5.9.0 - 5 Januari 2015

- Added AStyle integration.
- Remodeled Help >> About window.
- When looking up a piece of code by Ctrl+Clicking the destination line is centered.
- Removed caching from Tools >> Editor Options.
- Widened the exception window.
- Removed caching from the first-time configuration window.
- Improved parsing progress messages in the status bar.
- Improved performance of the classes list toolbar.
- Improved function declaration/definition switching.
- Significantly improved performance when selecting text.
- Fixed parser error when parsing C-style cast of string constant.
- Huge rewrite of code folding implementation.
- Major improvements in typing responsiveness.
- Major reduction in flickering during typing.
- Around 40-50% reduction in memory usage per open file.
- Fixed devcppPortable not passing files with spaces in names to devcpp correctly.
- Improved performance when opening multiple files from explorer using Dev-C++.
- Code completion does not show up anymore for periods typed as part of numbers.

Version 5.8.3 - 10 November 2014

- Slightly decreased flicker during editor opening.
- Makefiles are now edited as if they are source files. It's better than nothing.
- Opening braces after default are now completed correctly.
- Fixed a bug in NewFunctionFrm and NewVarFrm that caused crashes (thank you for reporting).
- Rewritten ancient source code of devcppPortable.exe. It is now immune to overflows due to arguments of length more than 400.
- Fixed a bug in TCppParser that caused it to ignore project include paths.
- Fixed a crash in TCppTokenizer due to spaces before #include in combination with comments after the <file> or "file" part (like " #include <foo> // bar").

Version 5.8.2 - 1 November 2014

- Fixed bug in procedure TMainForm.EditorSaveTimer that corrupted the editor views.
- Updated provided commit command.

Version 5.8.1 - 28 Oktober 2014

- Fixed an overflow error in TCppParser.CheckForStructs that caused errors when opening/saving files.
- Fixed a crash in TEditorList.GetFocusedPageControl that fixes random crashes.
- Refactored symbol completion code.
- The code completion window does not show anymore when the cursor changes during the timer rundown after typing . or :: or ->.
- Improved multiple file opening performance.
- Improved project closing performance.
- Updated compilation readme.

Version 5.8.0 - 21 Oktober 2014

- Added support for two file views next to each other.
- All windows are now opened at the centre of the main window instead of at random places across random monitors.
- Slightly improved performance of the function tooltip.
- Fixed a bug in makefile creation of C DLL's that prevented compilation.
- When compiling, header files are now treated separately from other files.
- Removed CVS support. CVS is considered deprecated, we recommend using explorer based code managers for alternatives like Git or SVN.

SOURCE 5.11/Source/AboutFrm.pas
{
    This file is part of Dev-C++
    Copyright (c) 2004 Bloodshed Software

    Dev-C++ is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    Dev-C++ is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Dev-C++; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
}

unit AboutFrm;

interface

uses
  Windows, SysUtils, Classes, version, Graphics, Controls, Forms, Dialogs,
  StdCtrls, Buttons, ExtCtrls;

type
  TAboutForm = class(TForm)
    lblVersion: TLabel;
    btnOk: TBitBtn;
    lblCopyright: TLabel;
    grpLicense: TGroupBox;
    txtLicense: TMemo;
    grpLinks: TGroupBox;
    lblBlood: TLabel;
    lblBloodURL: TLabel;
    lblMinGW: TLabel;
    lblMinGWURL: TLabel;
    lblDiscussion: TLabel;
    lblDiscussionURL: TLabel;
    lblMailing: TLabel;
    lblMailingURL: TLabel;
    lblResources: TLabel;
    lblResourcesURL: TLabel;
    lblBlog: TLabel;
    lblBlogURL: TLabel;
    btnAuthors: TBitBtn;
    btnUpdateCheck: TBitBtn;
    timerFish: TTimer;
    pnlFish: TPanel;
    FishImage: TImage;
    imgBanner: TImage;
    bvNew: TBevel;
    lblSubreddit: TLabel;
    lblSubredditURL: TLabel;
    lblRepository: TLabel;
    lblRepositoryURL: TLabel;
    lblTDM: TLabel;
    lblTDMURL: TLabel;
    lblPost4992: TLabel;
    lblCompilers: TLabel;
    lblPre4992: TLabel;
    bvCompilers: TBevel;
    imgDonate: TImage;
    procedure URLLabelClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure btnAuthorsClick(Sender: TObject);
    procedure btnAuthorsDragOver(Sender, Source: TObject; X, Y: Integer; State: TDragState; var Accept: Boolean);
    procedure btnAuthorsDragDrop(Sender, Source: TObject; X, Y: Integer);
    procedure timerFishTimer(Sender: TObject);
    procedure FishImageClick(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure imgDonateClick(Sender: TObject);
  private
    procedure LoadText;
  end;

implementation

uses
  ShellAPI, devcfg, utils, MultiLangSupport, main;

{$R *.dfm}

procedure TAboutForm.LoadText;
begin
  // Set interface font
  Font.Name := devData.InterfaceFont;
  Font.Size := devData.InterfaceFontSize;

  Caption := Lang[ID_AB_CAPTION];
  grpLicense.Caption := Lang[ID_AB_LICENSE];
  grpLinks.Caption := Lang[ID_AB_WEBCAP];
  lblPost4992.Caption := Lang[ID_AB_POST4992];
  lblRepository.Caption := Lang[ID_AB_REPOSITORY];
  lblSubreddit.Caption := Lang[ID_AB_SUBREDDIT];
  lblBlog.Caption := Lang[ID_AB_BLOGNAME];
  lblCompilers.Caption := Lang[ID_AB_COMPILERS];
  lblMinGW.Caption := Lang[ID_AB_LBLMINGWSITE];
  lblTDM.Caption := Lang[ID_AB_TDMGCC];
  lblPre4992.Caption := Lang[ID_AB_PRE4992];
  lblBlood.Caption := Lang[ID_AB_LBLBLOODSITE];
  lblDiscussion.Caption := Lang[ID_AB_LBLFORUM];
  lblMailing.Caption := Lang[ID_AB_LBLMAIL];
  lblResources.Caption := Lang[ID_AB_LBLEMAIL];
  btnOk.Caption := Lang[ID_BTN_OK];
  btnUpdateCheck.Caption := Lang[ID_AB_UPDATE];
  btnAuthors.Caption := Lang[ID_BTN_AUTHOR];
end;

procedure TAboutForm.URLLabelClick(Sender: TObject);
var
  S: AnsiString;
begin
  S := TLabel(Sender).Caption;
  ShellExecute(GetDesktopWindow(), 'open', PAnsiChar(s), nil, nil, SW_SHOWNORMAL);
end;

procedure TAboutForm.FormCreate(Sender: TObject);
var
  datestring: AnsiString;
begin
  LoadText;

  // Use modified time, not the PE headers
  datestring := GetBuildTime(ParamStr(0));
  lblVersion.Caption := lblVersion.Caption + DEVCPP_VERSION + #13#10 + 'Build time: ' + datestring;

  if FileExists(devData.Splash) then // TODO: check all folders?
    imgBanner.Picture.LoadFromFile(devData.Splash);
end;

procedure TAboutForm.btnAuthorsClick(Sender: TObject);
const
  MessageText =
    'Authors:'#13#10#13#10 +
    '- Post-4.9.9.2 development: Johan Mes'#13#10 +
    '- Development: Colin Laplace, Mike Berg, Hongli Lai, Yiannis Mandravellos'#13#10 +
    '- Contributors: Peter Schraut, Marek Januszewski, Anonymous'#13#10 +
    '- MinGW compiler system: Mumit Khan, J.J. van der Heijden, Colin Hendrix and GNU developers'#13#10 +
    '- Splash screen and association icons: Matthijs Crielaard'#13#10 +
    '- New Look theme: Gerard Caulfield'#13#10 +
    '- Gnome icons: Gnome designers'#13#10 +
    '- Blue theme: Thomas Thron'#13#10 +
	'- ESP32 esp-idf: rudi ;-)';
begin
  MessageBeep($F);
  MessageDlg(MessageText, MtInformation, [MbOK], 0);
end;

procedure TAboutForm.btnAuthorsDragOver(Sender, Source: TObject; X,
  Y: Integer; State: TDragState; var Accept: Boolean);
begin
  Accept := True;
end;

procedure TAboutForm.btnAuthorsDragDrop(Sender, Source: TObject; X,
  Y: Integer);
begin
  pnlFish.Left := 0 - pnlFish.Width;
  pnlFish.Top := Random(Height - pnlFish.Height - (pnlFish.Height div 3));
  pnlFish.Tag := 0;
  timerFish.Enabled := True;
end;

procedure TAboutForm.timerFishTimer(Sender: TObject);
begin
  if ((pnlFish.Tag = 0) and (pnlFish.Left > Width + 10)) or
    ((pnlFish.Tag <> 1) and (pnlFish.Left < 0 - pnlFish.Width - 10)) then
    timerFish.Enabled := False;

  if pnlFish.Tag = 0 then
    pnlFish.Left := pnlFish.Left + 5
  else
    pnlFish.Left := pnlFish.Left - 5;
  pnlFish.Top := pnlFish.Top + 1;
end;

procedure TAboutForm.FishImageClick(Sender: TObject);
begin
  pnlFish.Tag := not pnlFish.Tag;
end;

procedure TAboutForm.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  Action := caFree;
end;

procedure TAboutForm.imgDonateClick(Sender: TObject);
begin
    ShellExecute(GetDesktopWindow(), 'open',
    PAnsiChar('https://github.com/ESP32DE/devC_esp-idf'), nil, nil,
    SW_SHOWNORMAL);
end;

end.



SOURCE 5.11/Source/Version.pas
{
    This file is part of Dev-C++
    Copyright (c) 2004 Bloodshed Software

    Dev-C++ is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    Dev-C++ is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with Dev-C++; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
}

unit Version;

interface

//path delimiter
const
{$IFDEF WIN32}
  pd = '\';
{$ENDIF}
{$IFDEF LINUX}
  pd = '/';
{$ENDIF}

  // exe properties
  DEVCPP = 'Dev-C++';
  DEVCPP_VERSION = '5.11.esp-v1 by rudi ;-)';
  // DEVCPP_VERSION = '5.11';

  // delimiters
  DEV_INTERNAL_OPEN = '$__DEV_INTERNAL_OPEN';

  // config file names
  DEV_DEFAULTCODE_FILE = 'defaultcode.ini';
  DEV_SHORTCUTS_FILE = 'shortcuts.ini';
  DEV_CLASSFOLDERS_FILE = 'classfolders.ini';
  DEV_CODEINS_FILE = 'codeinsertion.ini';
  DEV_WEBMIRRORS_FILE = 'webmirrors.ini';
  DEV_TOOLS_FILE = 'tools.ini';

  // make stuff
  DEV_MAKE_FILE = 'Makefile.win';

  // themes
  DEV_GNOME_THEME = 'Gnome';
  DEV_NEWLOOK_THEME = 'New Look';
  DEV_BLUE_THEME = 'Blue';
  DEV_INTERNAL_THEME = 'New Look';

  LANGUAGE_DIR = 'Lang' + pd;
  ICON_DIR = 'Icons' + pd;
  HELP_DIR = 'Help' + pd;
  TEMPLATE_DIR = 'Templates' + pd;
  THEME_DIR = 'Themes' + pd;
  PACKAGES_DIR = 'Packages' + pd;

  // file extensions
  LIB_EXT = '.a';
  OBJ_EXT = '.o';
  DLL_EXT = '.dll';
  EXE_EXT = '.exe';
  DEV_EXT = '.dev';
  HTML_EXT = '.html';
  RTF_EXT = '.rtf';
  TEX_EXT = '.tex';
  INI_EXT = '.ini';
  TEMPLATE_EXT = '.template';
  SYNTAX_EXT = '.syntax';
  C_EXT = '.c';
  CPP_EXT = '.cpp';
  CC_EXT = '.cc';
  CXX_EXT = '.cxx';
  CP2_EXT = '.c++';
  CP_EXT = '.cp';
  H_EXT = '.h';
  HPP_EXT = '.hpp';
  RC_EXT = '.rc';
  RES_EXT = '.res';
  RH_EXT = '.rh';
  GCH_EXT = '.gch'; // precompiled header
  DEF_EXT = '.def'; // definitions file

  // program defaults
  MAKE_PROGRAM = 'mingw32-make.exe';
  GCC_PROGRAM = 'gcc.exe';
  GPP_PROGRAM = 'g++.exe';
  GDB_PROGRAM = 'gdb.exe';
  GDB32_PROGRAM = 'gdb32.exe';
  WINDRES_PROGRAM = 'windres.exe';
  GPROF_PROGRAM = 'gprof.exe';
  PACKMAN_PROGRAM = 'packman.exe';
  CLEAN_PROGRAM = 'rm.exe'; // cannot use del, it doesn't have a quiet mode

  // File dialog filters
  FLT_ALLFILES = 'All files (*.*)|*.*|';
  FLT_TEXTS = 'Text files (*.txt)|*.txt';
  FLT_LIBRARIES = 'Libraries (*.lib;*.a)|*.lib;*.a';
  FLT_PROJECTS = 'Dev-C++ project (*.dev)|*.dev';
  FLT_HEADS = 'Header files (*.h;*.hpp;*.rh;*.hh)|*.h;*.hpp;*.rh;*.hh';
  FLT_CS = 'C source files (*.c)|*.c';
  FLT_CPPS = 'C++ source files (*.cpp;*.cc;*.cxx;*.c++;*.cp)|*.cpp;*.cc;*.cxx;*.c++;*.cp';
  FLT_RES = 'Resource scripts (*.rc)|*.rc';
  FLT_MSVCPROJECTS = 'MS Visual C++ projects (*.dsp)|*.dsp';
  FLT_CBPROJECTS = 'Code::Blocks projects (*.cbp)|*.cbp';

  // Custom synedit style properties
  cBP = 'Breakpoints';
  cErr = 'Error line';
  cABP = 'Active breakpoints';
  cGut = 'Gutter';
  cSel = 'Selected text';
  cFld = 'Folding lines';

  // GPROF commands and displays
  GPROF_CHECKFILE = 'gmon.out';
  GPROF_CMD_GENFLAT = '-p';
  GPROF_CMD_GENMAP = '-q';

implementation

end.

